File: src/command/soldier.go
package command

import "fmt"

type Soldier struct {
	Name string
}

func (soldier Soldier) ExecCommand(command Command) {
	fmt.Print("I am " + soldier.Name + ", I will execute general's command: ")
	command.exec()
}

func (soldier Soldier) UndoCommand(command Command) {
	fmt.Print("I am " + soldier.Name + ", I will execute general's command: ")
	command.undo()
}

File: java/src/pattern/command/Soldier.java
package pattern.command;

public class Soldier {
    private String name;
    private Order command;

    public Soldier(String name) {
        this.name = name;
    }

    public void execCommand() {
        System.out.println(name + " " + command.execute());
    }

    public void undoCommand() {
        System.out.println(name + " " + command.undo());
    }

    public void setCommand(Order command) {
        this.command = command;
    }
}
